tools/wake-format/emitter.cpp

```diff
@@ -45,23 +45,35 @@
     }                                                                           \
   }
 
+bool Emitter::fits(const wcl::doc_builder& bdr, ctx_t ctx, wcl::doc doc) {
+  if (bdr.has_newline()) {
+    return bdr.last_width() + doc.first_width() <= max_column_width;
+  } else {
+    return bdr.last_width() + doc.first_width() + ctx.width <= max_column_width;
+  }
+}
+
+bool Emitter::requires_nl(ctx_t ctx, CSTElement node) {
+  return node.id() == CST_BLOCK && node.firstChildElement().id() != CST_MATCH;
```

Comment:
Question not a request: Do matches get wrapped in blocks?

Reply:
Oh, interesting point. Looking at my old code, the implication is no. I'll confirm that and fix this bit of code when I get to implementing walk_match